server/launcher/launcher.go

```diff
@@ -0,0 +1,179 @@
+	}
+
+	queryMap, accelerate, err := svc.tls.GetDistributedQueries(newCtx)
+	if err != nil {
+		return nil, false, errors.Wrap(err, "get queries for launcher")
+	}
+
+	result := &distributed.GetQueriesResult{
+		Queries:           queryMap,
+		AccelerateSeconds: int(accelerate),
+	}
+
+	return result, false, nil
+}
+
+func (svc *launcherWrapper) PublishLogs(ctx context.Context, nodeKey string, logType logger.LogType, logs []string) (string, string, bool, error) {
+	newCtx, invalid, err := svc.authenticateHost(ctx, nodeKey)
+	if err != nil {
+		return "", "", invalid, errors.Wrap(err, "authenticate launcher")
+	}
+
+	switch logType {
+	case logger.LogTypeStatus:
+		var statuses []kolide.OsqueryStatusLog
+		for _, log := range logs {
+			// StatusLog handles osquery logging messages
+			var statusLog = struct {
+				Severity string `json:"s"`
+				Filename string `json:"f"`
+				Line     string `json:"i"`
+				Message  string `json:"m"`
+			}{}
+
+			if err := json.NewDecoder(bytes.NewBufferString(log)).Decode(&statusLog); err != nil {
+				return "", "", false, errors.Wrap(err, "decode status log from launcher")
+			}
+
+			statuses = append(statuses, kolide.OsqueryStatusLog{
+				Severity: statusLog.Severity,
+				Filename: statusLog.Filename,
+				Line:     statusLog.Line,
+				Message:  statusLog.Message,
+			})
+		}
+
+		err = svc.tls.SubmitStatusLogs(newCtx, statuses)
+		return "", "", false, errors.Wrap(err, "submit status logs from launcher")
+	case logger.LogTypeSnapshot, logger.LogTypeString:
+		var results []kolide.OsqueryResultLog
+		for _, log := range logs {
+			var result kolide.OsqueryResultLog
+			if err := json.Unmarshal([]byte(log), &result); err != nil {
+				return "", "", false, errors.Wrap(err, "unmarshaling result log")
+			}
+			results = append(results, result)
+		}
+		err = svc.tls.SubmitResultLogs(newCtx, results)
+		return "", "", false, errors.Wrap(err, "submit result logs from launcher")
+	default:
+		// We have a logTypeAgent which is not there in the osquery-go enum.
+		// See https://github.com/kolide/launcher/issues/183
+		panic(fmt.Sprintf("%s log type not implemented", logType))
+	}
+}
+
+func (svc *launcherWrapper) PublishResults(ctx context.Context, nodeKey string, results []distributed.Result) (string, string, bool, error) {
+	newCtx, invalid, err := svc.authenticateHost(ctx, nodeKey)
+	if err != nil {
+		return "", "", invalid, err
+	}
+
+	osqueryResults := make(kolide.OsqueryDistributedQueryResults, len(results))
+	statuses := make(map[string]string, len(results))
+
+	for _, result := range results {
+		statuses[result.QueryName] = strconv.Itoa(result.Status)
+		osqueryResults[result.QueryName] = result.Rows
+	}
+
+	err = svc.tls.SubmitDistributedQueryResults(newCtx, osqueryResults, statuses)
+	return "", "", false, errors.Wrap(err, "submit launcher results")
+}
+
+func (svc *launcherWrapper) CheckHealth(ctx context.Context) (int32, error) {
+	healthy := health.CheckHealth(svc.logger, svc.healthCheckers)
```

Comment:
Is this a different health check interface than the one we were looking at yesterday?

Reply:
I merged and rebased the `health` package yesterday. This line calls the code you reviewed yesterday. A similar call is made in the http version.

Reply:
Ah I see, the `int32` in this function is compatible with the [gRPC definition](https://github.com/kolide/agent-api/blob/master/agent_api.proto#L158). Maybe we should kill the unused `error` return?

Reply:
The error return seems necessary because the go grpc generated code always has an error struct, but I'll add it for discussion in the launcher package.